File: Problem Solving/rotate_array.java
import java.util.*;
class solutions
{
	public static void main(String args[]){
		int a[]=new int[]{1,2,3};
		int k=9;
		new Solution().rotate(a,k);
	}
}
class Solution {
    public void rotate(int[] nums, int k) {
        if(nums.length==1){
            nums=Arrays.copyOf(nums,nums.length);
        }
        else if(nums.length<=k){
			final int rotated[]=Arrays.copyOf(nums,nums.length);
			int j=1;
            for(int i=0;i<k;i++){
				if(j==nums.length){
					j=0;
				}
				rotatedfork(nums,j,rotated);
				j+=1;
			}
        }
        else{
        int rotated[]=Arrays.copyOf(nums,nums.length);
        int index=0;
        for(int i=rotated.length-k;i<rotated.length;i++){
            nums[index]=rotated[i];
            index+=1;
        }
        for(int i=0;i<rotated.length-k;i++){
            nums[index]=rotated[i];
            index+=1;
        }
        }
    }
	public void rotatedfork(int[] nums,int k,int[] rotated){
		System.out.println(k);
        int index=0;
        for(int i=rotated.length-k;i<rotated.length;i++){
            nums[index]=rotated[i];
            index+=1;
        }
        for(int i=0;i<rotated.length-k;i++){
            nums[index]=rotated[i];
            index+=1;
        }
		for(int i=0;i<nums.length;i++){
			System.out.print(nums[i]+" ");
		}
		System.out.println();
	}
}

File: Problem Solving/rotate_Array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        length=len(nums)
        if length == 1:
            return
        if k == 0:
            return
        k=k%length
        i=0
        j=length-1
        while i<=j:
            nums[i],nums[j]=nums[j],nums[i]
            i+=1
            j-=1
        position=(k-1)
        i=0
        while i<=position:
            nums[i],nums[position]=nums[position],nums[i]
            i+=1
            position-=1
        
        position=k
        j=length-1
        while position<=j:
            nums[position],nums[j]=nums[j],nums[position]
            j-=1
            position+=1
        
        
        
        